### PR TITLE
Converted more uorb messages to autogen and worked on eff-c++ warnings.

### PR DESCRIPTION
**Pull request by jgoppert**
Also added ctor and virtual dtor to c++ autogen classes to silence -Weffc++ warnings. Still needs to be tested more, but wanted to get some feedback.

### CODE REVIEW

**Comment by LorenzMeier** [comment]
It looks good to me. I think we should work hard to validate fast and get this in, since its welcome cleanup but creates noise in the development flow if not done swiftly and smoothly. Did you flight test this change?

File: src/modules/sensors/sensors.cpp [hunk 1]
```diff
@@ -2052,35 +2052,24 @@ Sensors::task_main()
...
-	if (ret) {
-		goto exit_immediate;
-	}
-
-	ret = mag_init();
-
-	if (ret) {
-		goto exit_immediate;
+	int ret = 0;
+	for (int i=0; i<1; i++)  {
```
Comment (vooon) [inline src/modules/sensors/sensors.cpp]: Why you do this? Goto's here looks more readable for me. But i'm not fw dev.

File: src/modules/uORB/topics/filtered_bottom_flow.h [hunk 2]
```diff
@@ -33,34 +31,48 @@
  *
  ****************************************************************************/
 
-/**
- * @file filtered_bottom_flow.h
- * Definition of the filtered bottom flow uORB topic.
- */
+/* Auto-generated by genmsg_cpp from file /home/jgoppert/git/px4/Firmware/msg/filtered_bottom_flow.msg */
```
Comment (vooon) [inline src/modules/uORB/topics/filtered_bottom_flow.h]: Not sure, but i think auto-generated files should be removed. I think that simple rebuild should not lead to file changes in git.

Comment (LorenzMeier) [inline src/modules/uORB/topics/filtered_bottom_flow.h]: Yes, absolutely, I missed that one in my earlier review. No generated files in GIT, of course!

**Comment by jgoppert** [comment]
I haven't flight tested this yet. I will make sure to clean up the autogenerated files, this is kind of rough right now. I am investigating some odd glitches in the sensor_combined message with the debugger. Possibly the different layout of the c and c++ structs are causing issues. Any ideas?

odd_sensor_combined

File: src/modules/sensors/sensors.cpp (hunk 1 above)
Comment (jgoppert) [inline src/modules/sensors/sensors.cpp]: -Weffc++ doesn't allow goto

**Comment by jgoppert** [comment]
The C++ struct with virtual dtors were changing the memory layout so that they didn't match the C struct. So I'm not really sure I can make the code -Weffc++ compliant as is. I just reverted that change. It looks good in memory now and I will flight test in a couple of hours.

**Comment by LorenzMeier** [comment]
So this is now tested and would be good to go?

**Comment by jgoppert** [comment]
Yes, I have flight tested this on two systems with one running ROS and all the messages were fine and I didn't have any other issues or see anything when looking at the log files.

File: src/modules/sensors/sensors.cpp (hunk 1 above)
Comment (LorenzMeier) [inline src/modules/sensors/sensors.cpp]: do {} while(0) seems the more common pattern to do this.

File: src/drivers/sf0x/sf0x.cpp [hunk 3]
```diff
@@ -262,7 +262,7 @@ SF0X::init()
 {
 	/* do regular cdev init */
 	if (CDev::init() != OK) {
-		goto out;
```
Comment (LorenzMeier) [inline src/drivers/sf0x/sf0x.cpp]: Are we freeing all allocated fds correctly? (before the change and in particular after?)

Comment (jgoppert) [inline src/drivers/sf0x/sf0x.cpp]: The fds were not being closed in the existing code and my code kept the same behaviour. I'm submitting a commit now to fix it to always close.

**Comment by jgoppert** [comment]
Can someone test the change to the sf0x driver and make sure this clean up didn't cause any functional change. I don't have the hardware.

**Comment by LorenzMeier** [comment]
@severinleuenberger could you give this branch a try and report back if the laser driver is still working? I'd like to avoid breaking it for you, and the unit is not in widespread use currently.

**Pull request closed without merging**